### PR TITLE
Added the relevant target framework for project references in the lockfile

### PR DESCRIPTION
**Pull request by davidfowl**
- Only a restore change for now so we can have project files with
the framework be generated before taking advantage of it in the runtime.

#2622 

/cc @anurse @BillHiebert @jasonmalinowski @yishaigalatzer

### CODE REVIEW

File: src/Microsoft.Dnx.Tooling/Restore/RestoreCommand.cs [hunk 1]
```diff
@@ -862,14 +856,11 @@ private Task<LockFile> ReadLockFile(string projectLockFilePath)
                     var packageInfo = repository.FindPackagesById(library.Name)
                                                 .FirstOrDefault(p => p.Version == library.Version);
 
-                    var projectInfo = projectResolver.FindProject(library.Name);
-
-                    if (projectInfo != null)
+                    var projectDependency = projectResolver.FindProject(library.Name);
```
Comment (BrennanConroy) [inline src/Microsoft.Dnx.Tooling/Restore/RestoreCommand.cs]: Super nit: add newline here or remove it from same code above (L813)

**Comment by BrennanConroy** [comment]
:shipit:

File: test/Microsoft.Dnx.Tooling.FunctionalTests/DnuPublishTests.cs [hunk 2]
```diff
@@ -671,7 +671,8 @@ public void PublishMultipleProjectsWithDifferentTargetFrameworks(string flavor,
   ""targets"": {
     ""DNX,Version=v4.6"": {
       ""Lib/1.0.0"": {
-        ""type"": ""project""
+        ""type"": ""project"",
```
Comment (moozzyk) [inline test/Microsoft.Dnx.Tooling.FunctionalTests/DnuPublishTests.cs]: any tests for reading the lock file?

**Comment by anurse** [comment]
:shipit:

**Comment by davidfowl** [comment]
@mozzyz updated the lock file reading test

**Pull request merged**